src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -170,6 +170,7 @@ case class SupervisionRule(headName: String,
 case class InferenceRule(head: InferenceRuleHead,
                          q: ConjunctiveQuery,
                          weights: FactorWeight = null,
+                         valueExpr: Expr = null,
```

Comment:
Use of `null` in Scala is an anti-pattern.  `Option[...]` type not only makes it more wieldy but also safer from NPEs.  Looks like a few survived my past eradication attempt, but let's try not to introduce more. :)